src/main/kotlin/fandor/ReachEngineEventHandlerFunction.kt

```diff
@@ -17,46 +17,78 @@ import java.io.InputStream
 //import AWS from 'aws-sdk'
 
 @FunctionBean("reach-engine-event-handler")
-class ReachEngineEventHandlerFunction : Consumer< S3EventNotification> {
+class ReachEngineEventHandlerFunction : Consumer< S3EventNotification>,S3FileRepo,SNSTopicGateway {
```

Comment:
This is sort of a violation of the Single Responsibility Principle.  I would instead create new Implementation classes for the interfaces and inject them in the constructor:
```suggestion
class ReachEngineEventHandlerFunction(fileRepo: FileRepo, topicGateway: TopicGateway) : Consumer< S3EventNotification>{
```

Also, `SNS...` or `S3...` would be the name of the Implementing class not the interface.